source/binding/AssignmentExpressions.cpp

```diff
@@ -547,18 +550,35 @@ Expression& ConversionExpression::fromSyntax(Compilation& compilation,
     return *result;
 }
 
+void ConversionExpression::setCast(ConvertCast cast) {
+    ASSERT((cast_ == ConvertCast::Implicit || cast_ == ConvertCast::Explicit) &&
```

Comment:
I'm not sure why this logic is necessary. I think we should only determine the conversion kind when we create the ConversionExpression and then nobody should really be modifying it later.

Reply:
I remove this. After setting ConversionKind in construction, this kind of code is totally unnecessary.